hoomd/md/pytest/test_methods.py

```diff
@@ -3,393 +3,303 @@
+            list_length = len(get_value)
+            if parameter == 'S':
+                assert list_length == 6
+            for i in range(list_length):
+                assert get_value[i] == value[i]
+        else:
+            assert get_value == value
+
+    for parameter in method_base_params.extra_params:
+        value = method_base_params.extra_params[parameter]
+        assert getattr(integrator,parameter) == value
 
     type_A = hoomd.filter.Type(['A'])
-    nph.filter = type_A
-    assert nph.filter == type_A
+    integrator.filter = type_A
+    assert integrator.filter is type_A
+
+    for parameter in method_base_params.rejected_params:
+        value = method_base_params.rejected_params[parameter]
+        setattr(integrator,parameter,value)
+        assert getattr(integrator,parameter) == value
+
+    for parameter in method_base_params.accepted_params:
+        value = method_base_params.accepted_params[parameter]
+        setattr(integrator,parameter,value)
+        get_value = getattr(integrator,parameter)
+        if isinstance(value,list):
+            list_length = len(get_value)
+            if parameter == 'S':
+                assert list_length == 6
+            for i in range(list_length):
+                assert get_value[i] == value[i]
+        else:
+            assert get_value == value
+
+def test_attributes_attached(simulation_factory,
+                                two_particle_snapshot_factory,
+                                method_base_params):
 
-    ramp_s = [
-        hoomd.variant.Ramp(1.0, 4.0, 1000, 10000),
-        hoomd.variant.Ramp(2.0, 4.0, 1000, 10000),
-        hoomd.variant.Ramp(3.0, 4.0, 1000, 10000),
-        hoomd.variant.Ramp(0.125, 4.0, 1000, 10000),
-        hoomd.variant.Ramp(.25, 4.0, 1000, 10000),
-        hoomd.variant.Ramp(.5, 4.0, 1000, 10000)
-    ]
-    nph.S = ramp_s
-    assert len(nph.S) == 6
-    for i in range(6):
-        assert nph.S[i] is ramp_s[i]
+    all_ = hoomd.filter.All()
+    integrator = method_base_params.method(**method_base_params.setup_params,filter=all_)
 
-    nph.tauS = 10.0
-    assert nph.tauS == 10.0
+    sim = simulation_factory(two_particle_snapshot_factory())
+    sim.operations.integrator = hoomd.md.Integrator(0.005, methods=[integrator])
+    sim.run(0)
 
-    nph.box_dof = (True, False, False, False, True, False)
-    assert nph.box_dof == (True, False, False, False, True, False)
+    assert integrator.filter is all_
 
-    nph.couple = 'none'
-    assert nph.couple == 'none'
+    for parameter in method_base_params.setup_params:
+        value = method_base_params.setup_params[parameter]
+        get_value = getattr(integrator,parameter)
+        if isinstance(value,list):
+            list_length = len(get_value)
+            if parameter == 'S':
+                assert list_length == 6
+            for i in range(list_length):
+                assert get_value[i] == value[i]
+        else:
+            assert get_value == value
```

Comment:
It may be worth including a function like this as a helper since this check is done a lot.

```python
def check_instance_attrs(instance, attr_dict, set_attrs=False):
    for attr, value in attr_dict.items():
        if set_attrs:
            setattr(instance, attr, value)
        assert getattr(instance, attr) == value
```